pkg/conditions/manager_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2025 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/fake"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name string
+		want Manager
+	}{{
+		name: "New returns a non-nil manager.",
+		want: &managerImpl{},
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := New(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("New() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_conditionSet_Mark(t *testing.T) {
+	manager := New()
+
+	tests := []struct {
+		name  string
+		start []xpv1.Condition
+		mark  []xpv1.Condition
+		want  []xpv1.Condition
+	}{{
+		name:  "provide no conditions",
```

Comment:
Nit: Established pattern is to use a `CamelCase` subtest name and include a human-readable `reason` string that's printed if the test fails.

We also tend to use a `map[string]struct{}` where the string is the test name, but less worried about that.

I find this makes the output a lot easier to read when the test fails. Using the names to provide context like this Go'll mangle them to `provide_no_conditions`, `existing_status_a_new_condition_is_appended` etc. Not so bad here but it's not great when your test descriptions get a little longer.

Reply:
I gave it a try to implement as you mentioned